qiskit/transpiler/passes/optimization/template_matching/forward_match.py

```diff
@@ -412,6 +415,26 @@ def run_forward_match(self):
                     and self._is_same_op(node_circuit, node_template)
                 ):
 
+                # Check if parameters match the template or not.
+                    #Construct a temporary list of matches. 
+                    temp_match = self.match.copy()
+                    # Append the new match to the list of matches.
+                    temp_match.append([i,label])
+
+                    substitution = TemplateSubstitution(
+                        [temp_match],
+                        self.circuit_dag_dep,
+                        self.template_dag_dep,
+                        None
+                    )
+                    
+                    template_sublist = [x[0] for x in temp_match]
+                    circuit_sublist = [x[1] for x in temp_match]
+
+                    template = substitution._attempt_bind(template_sublist,circuit_sublist)
```

Comment:
You can refactor and combine the if statements by continuing if this bind is unsuccessful.